livedata/src/main/java/io/getstream/chat/android/livedata/EventBatchUpdate.kt

```diff
@@ -0,0 +1,109 @@
+package io.getstream.chat.android.livedata
+
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.livedata.extensions.updateLastMessage
+import io.getstream.chat.android.livedata.extensions.users
+
+/**
+ * EventBatchUpdate helps you efficiently implement a 4 step batch update process
+ * It updates multiple messages, users and channels at once.
+ *
+ * val batchBuilder = EventBatchUpdate.Builder()
+ *
+ * as a first step specify which channels and messages to fetch
+ * batchBuilder.addToFetchChannels()
+ * batchBuilder.addToFetchMessages()
+ *
+ * as a second step, load the required data for batch updating using
+ * val batch = batchBuilder.build(domainImpl)
+ *
+ * third, add the required updates via
+ * batch.addUser, addChannel and addMessage methods
+ *
+ * fourth, execute the batch using
+ * batch.execute()
+ */
+internal class EventBatchUpdate private constructor(
+    private val domainImpl: ChatDomainImpl,
+    channelMap: Map<String, Channel>,
+    messageMap: Map<String, Message>
+) {
+    private var users: Map<String, User> = emptyMap()
+    private var channels: Map<String, Channel> = channelMap
+    private var messages: Map<String, Message> = messageMap
+
+    internal class Builder {
+        private var channelsToFetch = setOf<String>()
+        private var messagesToFetch = setOf<String>()
+        private var users = setOf<User>()
+
+        fun addToFetchChannels(cIds: List<String>) {
+            channelsToFetch = channelsToFetch + cIds
+        }
+
+        fun addToFetchChannels(cId: String) {
+            channelsToFetch = channelsToFetch + listOf(cId)
+        }
+
+        fun addToFetchMessages(ids: List<String>) {
+            messagesToFetch = messagesToFetch + ids
+        }
+
+        fun addToFetchMessages(id: String) {
+            messagesToFetch = messagesToFetch + id
+        }
+
+        fun addUsers(usersToAdd: List<User>) {
+            users = users + usersToAdd
+        }
+
+        suspend fun build(domainImpl: ChatDomainImpl): EventBatchUpdate {
+            val messageMap: Map<String, Message> = domainImpl.repos.selectMessages(messagesToFetch.toList()).associateBy(Message::id)
```

Comment:
We used to create this `Map` by passing in users in addition to the messages, the users from all `UserEvent`s:

```kotlin
val messageMap = domainImpl.repos.messages.select(messagesToFetch.toList(), users.toMap()).associateBy { it.id }
```

We're now adding those users to the `users` set with `addUsers ` above, but not using them anywhere.